rust/template/tcp_channel/receiver.rs

```diff
@@ -39,22 +44,33 @@ enum Fd {
 }
 
 impl Fd {
+    fn shutdown(self, fd: RawFd) -> Result<(), Error> {
```

Comment:
Why does this method take `self`?

Reply:
Because the type implements `Copy` and Clippy kindly asks us to pass 8-byte-or-less values by value and not by reference.

Reply:
But you never actually use `self`in this function

Reply:
Well, it's a method so invocation happens through the object and it is hence implicitly available. The alternative is to declare it as `fn shutdown(fd: RawFd)` but that means invocation is awkward as `Self::shutdown(fd)` instead of `self.shutdown(fd)`. Or am I misunderstanding what you are saying?